### PR TITLE
added basic Authentication 

### PR DESCRIPTION
**Pull request by narharikale**
- added login feature
- added signup feature
- added logout feature

note: confirm password functionality not added yet

### CODE REVIEW

**Review by saketkum11 — APPROVED** [review]
- Clean Code
- Good use of words.
- Codes are understable with one go.

**Comment by rahulmistry751** [comment]
User is able to create a new account without filling any data on signup page. Please add the condition to check whether user has entered data inside input fields or not

File: src/components/navbar/navbar.jsx [hunk 1]
```diff
@@ -30,6 +39,7 @@ function Navbar() {
                             <span className="material-icons"> favorite_border </span>
                             <div className="badge-with-icon">3</div>
                         </Link>
+                        { isAuth ?  <button className=" m-0 btn-link color-gray-500" onClick={ () => logOutHandler() } >Logout</button> :<></> }
```
Comment (t007rushi) [inline src/components/navbar/navbar.jsx]: write null instead of ```<></>```

File: src/components/private-route/private-route.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Navigate } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
+
+
+ function PrivateRoute({ children }) {
+    const { isAuth } = useAuth();
+    return isAuth ? children : <Navigate to="/login" replace/>;
```
Comment (t007rushi) [inline src/components/private-route/private-route.js:7]: remove unwanted replace

File: src/context/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const authInitialState = localStorage.getItem("AuthToken") ? true : false
+
+
+
+
+function AuthProvider({ children }) {
+
+    const navigate = useNavigate();
+    const [isAuth, setIsAuth] = useState(authInitialState);
+    const [isErr , setIsErr] = useState(false)
```
Comment (t007rushi) [inline src/context/auth-context.js:19]: can have only one state for this

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            setIsAuth(true)
+            navigate('/')
+        } catch (error) {
+             console.error(error);
+             setIsErr(true);
+        }
+    };
+
+    const signupHandler = async (e, {firstName,lastName,email,password}) => {
+        e.preventDefault()
```
Comment (t007rushi) [inline src/context/auth-context.js]: can put this ```e.preventDefault()``` inside call in signupHandler & not pass in function as argument

File: src/pages/auth/login.jsx [hunk 5]
```diff
@@ -1,9 +1,75 @@
...
+                    <button
+                        type="submit"
+                        className="btn btn-primary font-size-regular w-100-per"
+                        >Login</button>
+                    <button
+                        className="btn btn-outline font-size-regular w-100-per"
+                        onClick={(e) => loginHandler(e, 'adarshbalika@gmail.com', 'adarshbalika')} >
+                        Login as a Guest</button>
+                    <Link to='/signup' className="color-gray-500 d-flex gap-sm">
+                        Create new account<span className="material-icons">arrow_forward</span>{" "}
```
Comment (t007rushi) [inline src/pages/auth/login.jsx:69]: remove spaces

File: src/pages/auth/login.jsx [hunk 6]
```diff
@@ -1,9 +1,75 @@
...
+        'email': "",
+        'password': "",
+    });
+
+    const [isPasswordVisible , setisPasswordVisible ] = useState(false);
+
+    return (
         <>
+            <div className="auth-form-container">
+                <form className="auth-form" onSubmit={(e) => loginHandler(e, user.email, user.password)}>
```
Comment (t007rushi) [inline src/pages/auth/login.jsx]: here call e.prevent default before call

**Review by t007rushi — COMMENTED** [review]
Just look for suggestion

**Review by rahulmistry751 — APPROVED** [review]
Functionalities are working fine

**Pull request merged**